### PR TITLE
Implement Container Managed Persistence with Deltaspike

### PR DESCRIPTION
**Pull request by MohamedFarouk-HMCTS**

### CODE REVIEW

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepositoryTest.java [hunk 1]
```diff
@@ -5,21 +5,25 @@
...
 import uk.gov.justice.services.example.cakeshop.persistence.entity.CakeOrder;
+import uk.gov.justice.services.test.utils.persistence.BaseTransactionalTest;
 
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
 import java.util.TimeZone;
 import java.util.UUID;
 
 import javax.inject.Inject;
+import javax.transaction.*;
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepositoryTest.java]: imports shouldn't be wild carded.

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java [hunk 2]
```diff
@@ -8,20 +8,23 @@
...
 
+import org.junit.After;
 import uk.gov.justice.services.example.cakeshop.persistence.entity.Ingredient;
+import uk.gov.justice.services.test.utils.persistence.BaseTransactionalTest;
 
 import java.util.List;
 import java.util.UUID;
 
 import javax.inject.Inject;
+import javax.transaction.*;
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java]: wild card

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java [hunk 3]
```diff
@@ -36,9 +39,11 @@
     private Ingredient ingredientA;
     private Ingredient ingredientB;
     private Ingredient ingredientC;
+    @Inject
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java]: shouldn't be injecting user transaction if it is extending the base class.

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java [hunk 4]
```diff
@@ -8,21 +8,24 @@
...
+
 import uk.gov.justice.services.example.cakeshop.persistence.entity.Recipe;
+import uk.gov.justice.services.test.utils.persistence.BaseTransactionalTest;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 import javax.inject.Inject;
+import javax.transaction.*;
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: wild card

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java [hunk 5]
```diff
@@ -39,18 +42,18 @@
...
         recipeRepository.save(recipeA);
         recipeB = createRecipe(RECIPE_ID_B, RECIPE_NAME_B, false);
         recipeRepository.save(recipeB);
-        recipeC = createRecipe(RECIPE_ID_C, RECIPE_NAME_C, true);
-        recipeRepository.save(recipeC);
     }
 
     @Test
-    public void shouldFindRecipeById() {
+    public void shouldFindRecipeById() throws SystemException, NotSupportedException, HeuristicRollbackException, HeuristicMixedException, RollbackException {
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: does this still throw all of these exceptions?

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java [hunk 6]
```diff
@@ -112,6 +115,36 @@ public void shouldReturnAllRecipes() {
...
+    @Test
+    public void removeTest() {
+        List<Recipe> recipeList = recipeRepository.findAll();
+
+        assertThat(recipeList, hasSize(3));
+        assertThat(recipeList, hasItems(recipeA));
+        assertThat(recipeList, hasItems(recipeB));
+        assertThat(recipeList, hasItems(recipeC));
+
+        Recipe recipeAA = recipeRepository.findBy(recipeA.getId());
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: we can move this to a private method, which will help avoid the variables names too.

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java [hunk 7]
```diff
@@ -112,6 +115,36 @@ public void shouldReturnAllRecipes() {
...
+        recipeRepository.remove(recipeBB);
+        Recipe recipeBBB = recipeRepository.findBy(recipeB.getId());
+        Assert.assertNull(recipeBBB);
+
+        Recipe recipeCC = recipeRepository.findBy(recipeC.getId());
+        recipeRepository.remove(recipeCC);
+        Recipe recipeCCC = recipeRepository.findBy(recipeC.getId());
+        Assert.assertNull(recipeCCC);
+
+
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: spacing

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java [hunk 8]
```diff
@@ -112,6 +115,36 @@ public void shouldReturnAllRecipes() {
...
+        List<Recipe> recipeList2 = recipeRepository.findAll();
```
Comment (matt-rich) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: suggest renaming the variable to remainingRecipes or something like that. Maybe even use initialRecipes as the first variable.

File: example-context/example-service/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApi.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import uk.gov.justice.services.core.annotation.Handles;
+import uk.gov.justice.services.core.annotation.ServiceComponent;
+import uk.gov.justice.services.core.sender.Sender;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import javax.inject.Inject;
+
+import static uk.gov.justice.services.core.annotation.Component.COMMAND_API;
+
+/**
```
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApi.java]: delete comments

File: example-context/example-service/example-command/example-command-api/src/raml/cakeshop-command-api.raml [hunk 10]
```diff
@@ -14,13 +14,19 @@ protocols: [ HTTP, HTTPS ]
         (mapping):
             requestType: application/vnd.cakeshop.add-recipe+json
             name: cakeshop.add-recipe
+
+        (mapping):
+            requestType: application/vnd.cakeshop.remove-recipe+json
+            name: cakeshop.remove-recipe
+
```
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-api/src/raml/cakeshop-command-api.raml]: delete whitespace line

File: example-context/example-service/example-command/example-command-api/src/test/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApiTest.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.junit.Test;
+
+import static uk.gov.justice.services.test.utils.core.helper.ServiceComponents.verifyPassThroughCommandHandlerMethod;
+
+
+public class RemoveRecipeCommandApiTest {
+
+    @Test
+    public void shouldHandleAddRecipeCommand() throws Exception {
```
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-api/src/test/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApiTest.java]: method name is incorrect 'add recipe'

File: example-context/example-service/example-command/example-command-controller/src/main/java/uk/gov/justice/services/example/cakeshop/command/controller/CakeShopCommandController.java [hunk 12]
```diff
@@ -11,22 +13,35 @@
...
 @ServiceComponent(COMMAND_CONTROLLER)
 public class CakeShopCommandController {
+    private static final Logger LOGGER = getLogger(CakeShopCommandController.class);
 
     @Inject
     Sender sender;
 
     @Handles("cakeshop.add-recipe")
     public void addRecipe(final JsonEnvelope command) {
+        LOGGER.info("=============> Inside add-recipe Command Handler");
```
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-controller/src/main/java/uk/gov/justice/services/example/cakeshop/command/controller/CakeShopCommandController.java]: not info level

File: example-context/example-service/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/RemoveRecipeCommandHandler.java [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+        final UUID recipeId = getUUID(command.payloadAsJsonObject(), FIELD_RECIPE_ID).get();
+
+        final EventStream eventStream = eventSource.getStreamById(recipeId);
+        final Recipe recipe = aggregateService.get(eventStream, Recipe.class);
+
+        eventStream.append(
+                recipe.removeRecipe(recipeId)
+                        .map(enveloper.withMetadataFrom(command)));
+    }
+
```
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/RemoveRecipeCommandHandler.java]: extra blank lines to be deleted.

File: example-context/example-service/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListener.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    RecipeRepository recipeRepository;
+
+    @Handles("cakeshop.recipe-removed")
+    public void recipeRemoved(final JsonEnvelope event) {
+        final String recipeId = event.payloadAsJsonObject().getString(FIELD_RECIPE_ID);
+        LOGGER.trace("=============> Inside remove-recipe Event Listener about to find recipeId: " + recipeId);
+        Recipe recipe1 = recipeRepository.findBy(UUID.fromString(recipeId));
+        LOGGER.trace("=============> Found remove-recipe Event Listener. RecipeId: " + recipe1);
+        recipeRepository.remove(recipe1);
+        Recipe recipe2 = recipeRepository.findBy(UUID.fromString(recipeId));
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListener.java]: we're not doing anything with this second value, should we assert that it is null ?

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepositoryTest.java (hunk 1 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepositoryTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.justice.services.example.cakeshop.persistence.RecipeRepository;
+import uk.gov.justice.services.example.cakeshop.persistence.entity.Recipe;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import javax.json.JsonObject;
+
+import java.util.UUID;
+
+import static org.mockito.Mockito.*;
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: wild card

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java (hunk 2 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java]: Done

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java (hunk 3 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/IngredientRepositoryTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    @Mock
+    private JsonObject payload;
+
+    @InjectMocks
+    private RecipeRemovedEventListener recipeRemovedEventListener;
+
+    @Before
+    public void setup() {
+        when(envelope.payloadAsJsonObject()).thenReturn(payload);
+        String recipeId =UUID.randomUUID().toString();
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: spacing on = sign

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java (hunk 5 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: Removed

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+        when(envelope.payloadAsJsonObject().getString(anyString())).thenReturn(recipeId);
+        when(recipeRepository.findBy(UUID.fromString(recipeId))).thenReturn(recipe);
+    }
+
+    @Test
+    public void shouldHandleRecipeRemovedEvent() throws Exception {
+
```
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: whitespace

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java (hunk 6 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: Done

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java (hunk 7 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java [hunk 18]
```diff
@@ -1,59 +1,58 @@
...
+import org.junit.*;
+import org.junit.runners.MethodSorters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import uk.gov.justice.services.eventsourcing.repository.jdbc.eventlog.EventLog;
 import uk.gov.justice.services.example.cakeshop.it.util.ApiResponse;
 import uk.gov.justice.services.example.cakeshop.it.util.StandaloneEventLogJdbcRepository;
 import uk.gov.justice.services.example.cakeshop.it.util.TestProperties;
 
+import javax.jms.*;
```
Comment (matt-rich) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: wildcard imports and most changes are refactoring due to having a different code style. Suggest you reapply the code style now that you have the correct one.

File: example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java (hunk 8 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/test/java/uk/gov/justice/services/example/cakeshop/persistence/RecipeRepositoryTest.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java [hunk 19]
```diff
@@ -142,6 +154,86 @@ public void shouldRegisterRecipeAddedEvent() {
     }
 
     @Test
+    public final void singleAddAndRemoveRecipe() throws InterruptedException {
+        shouldRegisterRecipeRemovedEvent();
+    }
+
```
Comment (matt-rich) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: blank lines

File: example-context/example-service/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApi.java (hunk 9 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApi.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java [hunk 20]
```diff
@@ -142,6 +154,86 @@ public void shouldRegisterRecipeAddedEvent() {
...
+                .assertEquals("stream.id", recipeId)
+                .assertEquals("stream.version", 1);
+        String eventPayload = event.getPayload();
+        with(eventPayload)
+                .assertThat("$.recipeId", equalTo(recipeId))
+                .assertThat("$.name", equalTo("Vanilla cake"))
+                .assertThat("$.glutenFree", equalTo(false))
+                .assertThat("$.ingredients[0].name", equalTo("vanilla"))
+                .assertThat("$.ingredients[0].quantity", equalTo(2));
+        LOGGER.debug("Ending with ADD");
```
Comment (matt-rich) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Not sure this debugging comes out anywhere?

Consider if we need it or not (and the logger at the top if we don't)

File: example-context/example-service/example-command/example-command-api/src/raml/cakeshop-command-api.raml (hunk 10 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-command/example-command-api/src/raml/cakeshop-command-api.raml]: Done

File: example-context/example-service/example-command/example-command-api/src/test/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApiTest.java (hunk 11 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-command/example-command-api/src/test/java/uk/gov/justice/services/example/cakeshop/command/api/RemoveRecipeCommandApiTest.java]: Done

File: test-utils/test-utils-persistence/src/main/java/uk/gov/justice/services/test/utils/persistence/BaseTransactionalTest.java [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.UserTransaction;
+
+import org.junit.After;
+import org.junit.Before;
+
+public abstract class BaseTransactionalTest {
```
Comment (matt-rich) [inline test-utils/test-utils-persistence/src/main/java/uk/gov/justice/services/test/utils/persistence/BaseTransactionalTest.java]: javadoc - why do we use this class?

File: example-context/example-service/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/RemoveRecipeCommandHandler.java (hunk 13 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/RemoveRecipeCommandHandler.java]: Done

File: example-context/example-service/example-command/example-command-controller/src/main/java/uk/gov/justice/services/example/cakeshop/command/controller/CakeShopCommandController.java (hunk 12 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-command/example-command-controller/src/main/java/uk/gov/justice/services/example/cakeshop/command/controller/CakeShopCommandController.java]: Test pass through handler Works with Logger

File: example-context/example-service/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListener.java (hunk 14 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListener.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java (hunk 15 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java (hunk 16 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java (hunk 17 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/RecipeRemovedEventListenerTest.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java (hunk 18 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java (hunk 19 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Done

File: example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java (hunk 20 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-service/example-it/src/test/java/uk/gov/justice/services/example/cakeshop/it/CakeShopIT.java]: Done , removed logger

File: example-context/example-persistence/src/main/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepository.java [hunk 22]
```diff
@@ -1,13 +1,14 @@
...
+import org.apache.deltaspike.data.api.EntityManagerConfig;
 import uk.gov.justice.services.example.cakeshop.persistence.entity.CakeOrder;
 
 import java.util.UUID;
 
 import org.apache.deltaspike.data.api.EntityRepository;
 import org.apache.deltaspike.data.api.Repository;
 
-@Repository
+@Repository(forEntity = CakeOrder.class)
```
Comment (purple52) [inline example-context/example-persistence/src/main/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepository.java]: Is this pattern of the solution? Does it still work with this removed?

File: persistence/persistence-deltaspike/src/main/java/uk/gov/justice/services/persistence/EntityManagerProducer.java [hunk 23]
```diff
@@ -11,21 +13,23 @@
...
  */
+@ApplicationScoped
 public class EntityManagerProducer {
     private static final String UTC = "UTC";
 
     @PersistenceUnit
     EntityManagerFactory entityManagerFactory;
 
     @Produces
+    @RequestScoped
```
Comment (purple52) [inline persistence/persistence-deltaspike/src/main/java/uk/gov/justice/services/persistence/EntityManagerProducer.java]: How does this impact transactions started from MDBS? Does processing a message count aRequests coped?

Comment (purple52) [inline persistence/persistence-deltaspike/src/main/java/uk/gov/justice/services/persistence/EntityManagerProducer.java]: *MDBs *as RequestScoped  - sorry, autocorrect

Comment (MohamedFarouk-HMCTS) [inline persistence/persistence-deltaspike/src/main/java/uk/gov/justice/services/persistence/EntityManagerProducer.java]: Background: 
 Container Managed EntityManagerProducer as Application Scoped (1 for each war)
 Injected Entity Manager Factory (1 for each war) 
 Contextual (programatic) Entity manager (Request Scoped)   (1 for each MDB)
which creates request scoped PersistentContext and attaches a request scoped TransactionContext

Impact of Request Scope in terms of transaction context is active from the start to end of the MDB, yes so processing a message cound as a request scoped  transaction. 
Also transactional test (add and remove) passes now which was failing before using bean managed Persistence/transactions

File: example-context/example-persistence/src/main/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepository.java (hunk 22 above)
Comment (MohamedFarouk-HMCTS) [inline example-context/example-persistence/src/main/java/uk/gov/justice/services/example/cakeshop/persistence/CakeOrderRepository.java]: Yes removed as suggested and all tests successfully pass

**Pull request merged**